packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -501,6 +479,52 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
     return isEqualValues;
   }
 
+  #getCorrectNetworkClientIdByChainId(
+    chainIds: Hex[] | undefined,
+  ): { chainId: Hex; networkClientId: NetworkClientId }[] | null {
+    const { networkConfigurationsByChainId, selectedNetworkClientId } =
+      this.messagingSystem.call('NetworkController:getState');
+
+    if (!chainIds) {
+      const networkConfiguration = this.messagingSystem.call(
+        'NetworkController:getNetworkConfigurationByNetworkClientId',
+        selectedNetworkClientId,
+      );
+
+      if (networkConfiguration) {
+        return [
+          {
+            chainId: networkConfiguration.chainId,
+            networkClientId: selectedNetworkClientId,
+          },
+        ];
+      }
+      return null;
+    }
+
+    return chainIds.map((chainId) => {
```

Comment:
We don't want to consider `selectedNetworkClientId` or take the `[0]th` network client.  We should always take the one that's marked as the default.  So this should be something like

```
      const configuration = networkConfigurationsByChainId[chainId];
      return {
        chainId,
        networkClientId: configuration.rpcEndpoints[
          configuration.defaultRpcEndpointIndex].networkClientId,
      }
```

Reply:
resolved